exonum/src/crypto/mod.rs

```diff
@@ -687,22 +684,39 @@ mod tests {
     fn debug_format() {
         // Check zero padding
         let hash = Hash::new([1; HASH_SIZE]);
-        assert_eq!(format!("{:?}", &hash), "Hash(01010101)");
+        assert_eq!(format!("{:?}", &hash), "Hash(0101010101010101010101010101010101010101010101010101010101010101)");
 
         let pk = PublicKey::new([15; PUBLIC_KEY_LENGTH]);
-        assert_eq!(format!("{:?}", &pk), "PublicKey(0F0F0F0F)");
+        assert_eq!(format!("{:?}", &pk), "PublicKey(0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F0F)");
         let sk = SecretKey::new([8; SECRET_KEY_LENGTH]);
-        assert_eq!(format!("{:?}", &sk), "SecretKey(08080808...)");
+        assert_eq!(format!("{:?}", &sk), "SecretKey(08080808080808080808080808080808080808080808080808080808080808080808080808080808080808080808080808080808080808080808080808080808)");
         let signature = Signature::new([10; SIGNATURE_LENGTH]);
-        assert_eq!(format!("{:?}", &signature), "Signature(0A0A0A0A)");
+        assert_eq!(format!("{:?}", &signature), "Signature(0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A0A)");
         let seed = Seed::new([4; SEED_LENGTH]);
-        assert_eq!(format!("{:?}", &seed), "Seed(04040404...)");
+        assert_eq!(format!("{:?}", &seed), "Seed(0404040404040404040404040404040404040404040404040404040404040404)");
 
         // Check no padding
         let hash = Hash::new([128; HASH_SIZE]);
-        assert_eq!(format!("{:?}", &hash), "Hash(80808080)");
+        assert_eq!(format!("{:?}", &hash), "Hash(8080808080808080808080808080808080808080808080808080808080808080)");
         let sk = SecretKey::new([255; SECRET_KEY_LENGTH]);
-        assert_eq!(format!("{:?}", &sk), "SecretKey(FFFFFFFF...)");
+        assert_eq!(format!("{:?}", &sk), "SecretKey(FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)");
+    }
+
+    // Note that only public values have Display impl.
+    #[test]
+    fn display_format() {
+        // Check zero padding
+        let hash = Hash::new([1; HASH_SIZE]);
+        assert_eq!(format!("{}", &hash), "0101010101010101010101010101010101010101010101010101010101010101");
```

Comment:
Mmm... It seems invalid. `Display` should crop output to some amount of symbols then.

Reply:
[The docs] seem to suggest that `Display` should accurately represent the value as a string (i.e., so that one can write `from_str()` for that format). From this viewpoint we kinda should output the entire value there. But it can also be said that this is a use-case for `Serialize` trait, not Display.

[The docs]: https://doc.rust-lang.org/beta/std/fmt/index.html#fmtdisplay-vs-fmtdebug

Reply:
Huh? `Display` trait is for human convenience only, it has no reasons to be an accurate representation.